### PR TITLE
Feature/add missing publication endpoints

### PR DESCRIPTION
**Pull request by LHommeDeBat**
#### Short Description
Changes/Enhancements needed to implement linking/unlinking of algorithms and publications

#### Proposed Changes
  * Added more endpoints to PublicationController, allowing linking of algorithms and publications

**Fixes**: 
 * Some JPA annotations had 'joinColumn' and 'inverseJoinColumn' swapped
 * toModel-Method of EntityModel in GenericLinkAssembler now adds the hateoas links

### CODE REVIEW

File: org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/controller/PublicationController.java [hunk 1]
```diff
@@ -112,17 +125,52 @@
...
+        publicationMixIn.unlinkPublication(algorithm, id);
+        algorithmService.save(algorithm);
+        return new ResponseEntity<>(HttpStatus.OK);
+    }
+
+    @Operation(responses = {@ApiResponse(responseCode = "200"), @ApiResponse(responseCode = "400"), @ApiResponse(responseCode = "404")})
+    @GetMapping("/{id}/" + Constants.IMPLEMENTATIONS)
+    public HttpEntity<CollectionModel<EntityModel<ImplementationDto>>> getImplementations(@PathVariable UUID id) {
+        log.debug("Get implementations of Publication with id {}", id);
+        Set<Implementation> implementations = publicationService.findPublicationImplementations(id);
```
Comment (TGrosskopf) [inline org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/controller/PublicationController.java:170]: Why not use getImplementations() like in algorithms? See line 133

Comment (LHommeDeBat) [reply org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/controller/PublicationController.java:170]: Implementations is not yet implemented. This method was already there. I will add implementations in the coming days

File: org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/controller/PublicationController.java [hunk 2]
```diff
@@ -112,17 +125,59 @@
...
-//        Set<Algorithm> algorithms = publicationService.findPublicationAlgorithms(id);
-//        Set<AlgorithmDto> algorithmDtos = ModelMapperUtils.convertSet(algorithms, AlgorithmDto.class);
-//        CollectionModel<EntityModel<AlgorithmDto>> resultCollection = HateoasUtils.generateCollectionModel(algorithmDtos);
-//        algorithmAssembler.addLinks(resultCollection);
-//        publicationAssembler.addAlgorithmLink(resultCollection, id);
-//        return new ResponseEntity<>(resultCollection, HttpStatus.OK);
-//    }
+    @Operation(responses = {@ApiResponse(responseCode = "200"), @ApiResponse(responseCode = "400"), @ApiResponse(responseCode = "404")})
+    @GetMapping("/{id}/" + Constants.ALGORITHMS)
+    public HttpEntity<CollectionModel<EntityModel<AlgorithmDto>>> getAlgorithms(@PathVariable UUID id) {
```
Comment (FlxB2) [inline org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/controller/PublicationController.java:130]: Please change the method name to something like `getAlgorithmsFromPublication`. It's more descriptive and prevents collisions with operationIds. Same for the other added methods in this class. Alternatively, add operationId = "<someId>" to the @Operation annotation

Comment (LHommeDeBat) [reply org.planqk.atlas.web/src/main/java/org/planqk/atlas/web/controller/PublicationController.java]: Methods are renamed, in front- and backend

**Pull request merged**